core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1386,8 +1386,12 @@ class DAGScheduler(
               )
             }
           }
-          // Mark the map whose fetch failed as broken in the map stage
-          if (mapId != -1) {
+
+          if (mapStage.rdd.recomputeAllPartitionsOnFailure()) {
```

Comment:
hmmm, what if we have a `map` after `repartition`? then the root RDD will return false on `recomputeAllPartitionsOnFailure`